app/views/users/new.html.erb

```diff
@@ -1,3 +1,7 @@
+<% @user.errors.full_messages.each do |msg| %>
+<%= msg %>
+<% end %>
```

Comment:
High Level: Rails pipes missing/incorrect fields as errors into the User , which can then be printed on the View.